src/Simulation/TargetDefinitions/Decompositions/ApplyControlledX.qs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// Applies the controlled-X (or CNOT) gate to a pair of qubits. Note that the Controlled
+    /// functor is not supported.
+    ///
+    /// # Description
+    /// \begin{align}
+    ///     \operatorname{CNOT} \mathrel{:=}
+    ///     \begin{bmatrix}
+    ///         1 & 0 & 0 & 0 \\\\
+    ///         0 & 1 & 0 & 0 \\\\
+    ///         0 & 0 & 0 & 1 \\\\
+    ///         0 & 0 & 1 & 0
+    ///     \end{bmatrix},
+    /// \end{align}
+    ///
+    /// where rows and columns are ordered as in the quantum concepts guide.
+    ///
+    /// # Input
+    /// ## control
+    /// Control qubit for the CNOT gate.
+    /// ## target
+    /// Target qubit for the CNOT gate.
+    ///
+    /// # Remarks
+    /// Equivalent to:
+    /// ```qsharp
+    /// CNOT(control, target);
+    /// ```
+    @EnableTestingViaName("Test.TargetDefinitions.ApplyControlledX")
+    operation ApplyControlledX (control : Qubit, target : Qubit) : Unit is Adj {
```

Comment:
Totally agreed. It's definitely something I was planning on calling out as part of the feature branch merge into master (at the latest). In total between both this Type1 work and the previous Type2 work, there are 16 new public callables in the Microsoft.Quantum.Intrinsic namespace: ApplyControlledX, ApplyControlledZ, ApplyUncontrolledH, ApplyUncontrolledRx, ApplyUncontrolledRy, ApplyUncontrolledRz, ApplyUncontrolledS, ApplyUncontrolledT, ApplyUncontrolledX, ApplyUncontrolledY, ApplyUncontrolledZ, IsingXX, IsingYY, IsingZZ, CheckQubitUniqueness, and RotationAngleValidation. That's a lot of redundant callables! Of those, only the three Ising gates are ones I think have a decent argument to be public, and even there I'd be fine if we instead decided to keep them internal. But as I'll go into in the other comment thread, we don't have a good solution for extensible internal intrinsics.

Reply:
To summarize the temporary resolution we arrived at below, we are going to return to the question of the right way to balance extensibility vs making helper intrinsics public at a later date, and for now I've moved all 16 of these new callabes to be internal. That way they don't require an API review (yet). I've added this as a checklist item to #249 with the understanding that it may bet separated out into its own issue later.